libclamav/crypto.c

```diff
@@ -1758,16 +1759,19 @@ struct tm *cl_ASN1_GetTimeT(ASN1_TIME *timeobj)
     struct tm localtm;
 #endif
 
-    if (!(timeobj) || !(timeobj->data))
-        return NULL;
+    if (!(timeobj))
+        goto err;
+
+    if ((str = strndup(ASN1_STRING_get0_data(timeobj),
+                       ASN1_STRING_length(timeobj))) == NULL)
+        goto err;
 
-    str = (char *)(timeobj->data);
     if (strlen(str) < 12)
-        return NULL;
+        goto err;
 
     t = (struct tm *)calloc(1, sizeof(struct tm));
     if (!(t))
-        return NULL;
+        goto err;
 
     if (timeobj->type == V_ASN1_UTCTIME) {
```

Comment:
```suggestion
    if (ASN1_STRING_type(timeobj) == V_ASN1_UTCTIME) {
```

Similarly in the else if for `V_ASN1_GENERALIZEDTIME` a few lines down, which github doesn't let me add.